hls4ml/backends/vivado/passes/pointwise_codegen.py

```diff
@@ -0,0 +1,84 @@
+
+    Returns:
+        str: Generated C++ function
+    """
+
+    generated_code = (
+        "template<class data_T, class res_T, typename CONFIG_T>\n"
+        "class pointwise_conv_{index} : public PointwiseConv1D<data_T, res_T, CONFIG_T> {{\n"
+        "  public:\n"
+        "    static void pointwise_conv(\n"
+        "                               data_T data[CONFIG_T::in_width * CONFIG_T::n_chan],\n"
+        "                               res_T res[CONFIG_T::out_width * CONFIG_T::n_filt],\n"
+        "                               typename CONFIG_T::weight_t weights[CONFIG_T::n_chan * CONFIG_T::n_filt],\n"
+        "                               typename CONFIG_T::bias_t biases[CONFIG_T::n_filt]) {{\n"
+        "        data_T data_tmp[CONFIG_T::reuse_factor][CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor];\n"  # noqa: E501
+        "        #pragma HLS ARRAY_PARTITION variable=data_tmp complete dim=0\n"
+        "        res_T res_tmp[CONFIG_T::reuse_factor][CONFIG_T::out_width * CONFIG_T::n_filt / CONFIG_T::reuse_factor];\n"  # noqa: E501
+        "        #pragma HLS ARRAY_PARTITION variable=res_tmp complete dim=0\n\n"
+        "    RFInputLoop:\n"
+        "        for (int jj = 0; jj < CONFIG_T::reuse_factor; jj++) {{\n"
+        "        #pragma HLS UNROLL\n"
+        "        InnerInputLoop:\n"
+        "            for (int ii = 0; ii < CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor; ii++) {{\n"
+        "                #pragma HLS UNROLL\n"
+        "                data_tmp[jj][ii] = data[jj * CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor + ii];\n"  # noqa: E501
+        "            }}\n"
+        "        }}\n\n"
+    ).format(index=layer_idx)
+    indent = "        "
+    for i in range(reuse_factor):
+        generated_code += indent
+        generated_code += (
+            f"pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[{i}], res_tmp[{i}], weights, biases);\n"
+        )
+
+    generated_code += (
+        "\n"
+        "    RFOutputLoop:\n"
+        "        for (int jj = 0; jj < CONFIG_T::reuse_factor; jj++) {\n"
+        "        #pragma HLS UNROLL\n"
+        "        InnerOutputLoop:\n"
+        "            for (int ii = 0; ii < CONFIG_T::out_width * CONFIG_T::n_filt / CONFIG_T::reuse_factor; ii++) {\n"
+        "                #pragma HLS UNROLL\n"
+        "                res[jj * CONFIG_T::out_width * CONFIG_T::n_filt / CONFIG_T::reuse_factor + ii] = res_tmp[jj][ii];\n"  # noqa: E501
+        "            }\n"
+        "        }\n"
+        "    }\n"
+        "};\n"
+    )
+
+    return generated_code
+
+
+class GeneratePointwiseConv1D(OptimizerPass):
+    '''Generates code for pointwise 1D convolution'''
+
+    def match(self, node):
+        return isinstance(node, Conv1D) and node.model.config.get_config_value('IOType') == 'io_parallel'
+
+    def transform(self, model, node):
+        node_class = node.__class__.__name__
```

Comment:
Minor point, but we have `node.class_name` for this purpose, but wouldn't make a big difference in the check here.

The bigger question is why the check at all? In what cases can it fail?

Reply:
removed check in https://github.com/jmduarte/hls4ml/pull/22/commits/d56dc7349e94ad8b6aaefe5b0f22385d3d9a9e03